meetupHelpers.js

```diff
@@ -0,0 +1,52 @@
+const { joinArrayHumanReadable } = require('./utils');
+
+async function fetchNextEventForGroup (groupId) {
+  const fetch = await import('node-fetch');
+  
+  const variables = { groupId }
+  
+  const query = `query GetUpcomingEventsForGroup ($groupId: ID) {
+    group(id: $groupId) {
+      id,
+      name,
+      upcomingEvents (input: {first: 1}) {
+        edges {
+          node {
+            dateTime,
+            timezone,
+            shortUrl,
+            tickets {
+              edges {
+                node {
+                  user {
+                    name
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }`;
+  
+  const response = await fetch.default('https://api.meetup.com/gql', {
+    method: 'post',
+    body: JSON.stringify({query, variables}),
+    headers: {'Content-Type': 'application/json'}
+  });
+  
+  const data = await response.json();
+  
+  return data.data.group.upcomingEvents.edges[0].node;
```

Comment:
Is it possible for data.data, data.data.group, data.data.group.upcomingEvents ... to be empty at each level?

We want to return undefined/null or some object from this function on failure, but what do we do about errors from attempting to connect to meetup? eg. 500 error.

Reply:
I'm thinking I'll just wrap the whole thing in a try catch instead of having 8 '?.'s, and then just return undefined for anything other than a success, then we can try again a handful of times on failure.

Reply:
You might want to log the error so it can be debugged. console.error()?